### ISSUE TITLE
how to remove inserted record in test?

### ISSUE THREAD

**Issue by codeweb** [database-rider/database-rider#68]
I would now how to remove inserted record in a test environment:
```
@Test
@DataSet(transactional=true)
@ExpectedDataSet(value = "foo.csv", ignoreCols = "id")
public void runGenericTest() throws Exception {
```
Using CleanAfter/Before will remove records also in other tables.
Using the following code:
```
@After
public void clean_all() {
```
it will remove records before ExpectedDataSet action.

**Comment by rmpestano** [comment]
Hi, currently there is no way to clean only inserted records after test  execution. Also note that [seedStrategy](https://github.com/database-rider/database-rider/blob/master/rider-core/src/main/java/com/github/database/rider/core/api/dataset/SeedStrategy.java#L26) is applied on tables declared in @DataSet.

Having said that, maybe we could have a new seedStratagy e.g _CLEAN_INSERT_CLEAN_ which would clean records on tables declared in @DataSet after test execution.

**Comment by codeweb** [comment]
Is it possible to set tables on which CleanAfter/Before works?

**Comment by rmpestano** [comment]
Well, it uses the tableOrdering attribute in a first moment ([see here](https://github.com/database-rider/database-rider/blob/master/rider-core/src/main/java/com/github/database/rider/core/dataset/DataSetExecutorImpl.java#L490)) but after that it clears other tables also, [see here](https://github.com/database-rider/database-rider/blob/71ca3783a0dae20056fd68802651390b11dce618/rider-core/src/main/java/com/github/database/rider/core/dataset/DataSetExecutorImpl.java#L501).

Don't know how we could define that we won't clear other tables in a way to not confuse the api.  Example adding another attribute to @DataSet annotation like "skipAutomaticCleanup" may lead to confusion.  

So unless you have another idea I still think adding another seedingStragy is the recommended way.

**Comment by codeweb** [comment]
I agree and it's very interesting. 
Moreover I'm asking myself who can want delete all tables in database and why. Maybe "skipAutomaticCleanup" can help also in this situation.

**Comment by rmpestano** [comment]
Also another idea is to configure that behaviour (the "skipAutomaticCleanup") in @DBUnit annotation or in dbunit.yml see [configuration section here](https://github.com/database-rider/database-rider#configuration), but that will apply to all tests (in case of dbunit.yml) or in all tests of test class when using @DBunit annotation, something like:


@RunWith(JUnit4.class)
@DBUnit(automaticCleanUp = false) //we need to think in a better name
public class CrudIt {

@Test
@DataSet(transactional=true, cleanAfter=true, tableOrdering="foo,bar") //will clean only 'foo' and 'bar' tables 
@ExpectedDataSet(value = "foo.csv", ignoreCols = "id")
public void runGenericTest() throws Exception {
   }

@Test
@DBunit(automaticCleanUp = true)
@DataSet(transactional=true, cleanAfter=true) //will clean all tables because method level annotation takes precedence
@ExpectedDataSet(value = "foo.csv", ignoreCols = "id")
public void runGenericTest() throws Exception {
   }
}

What do you think?

**Comment by rmpestano** [comment]
"Moreover I'm asking myself who can want delete all tables in database and why" well usually the test database is used only for tests and an easy(and recommended) way to guarantee desired state is to clear everything before.

**Comment by codeweb** [comment]
Not everything but only interested tables... What do you think?

**Comment by rmpestano** [comment]
Yes, clean_insert strategy do that, clear tables declared in @Dataset("mydataset.yml") before test execution

**Comment by codeweb** [comment]
And what if dataset is not specified (I use only ExpectedDataSet)?

**Comment by rmpestano** [comment]
https://github.com/database-rider/database-rider/issues/68#issuecomment-359386258

**Comment by codeweb** [comment]
Ok, thanks. I hope someone will do it.

**Comment by rmpestano** [comment]
Let's keep it opened until there.

Thank you.
